internal/nginx/file/folders_test.go

```diff
@@ -0,0 +1,91 @@
+package file_test
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/nginxinc/nginx-kubernetes-gateway/internal/nginx/file"
+	"github.com/nginxinc/nginx-kubernetes-gateway/internal/nginx/file/filefakes"
+)
+
+func writeFile(t *testing.T, name string, data []byte) {
+	t.Helper()
+
+	//nolint:gosec
+	err := os.WriteFile(name, data, 0o644)
```

Comment:
inline (to save more comments like this, I'll just say do an audit of existing errors that you've added and update them)